docs/spec_draft.md

```diff
@@ -1444,6 +1446,170 @@ specification. Numeric precision is implementation-defined.
+| `scatter_dims_to_operand_dims` | 1-dimensional tensor constant of type `si64`      | (C6),(C7), (C8)                              |
+| `index_vector_dim`             | constant of type `si64`                           | (C5), (C6), (C14), (C15)                     |
+| `indices_are_sorted`           | constant of type `i1`                             |                                              |
+| `unique_indices`               | constant of type `i1`                             |                                              |
+| `update_computation`           | `function`                                        | (C13)                                        |
+
+### Outputs
+
+| Name      | Type                                              |
+|-----------|---------------------------------------------------|
+| `results` | variadic number of tensors of any supported types |
+
+### Constraints
+
+
+  * (C1) For any k $\in$ [0, N), rank(`inputs`[k]) = size(`update_window_dims`) +
+         size(`inserted_window_dims`).
+
+  * (C2) N $=$ size(`inputs`) == size(`updates`) and N $\ge$ 1.
+
+  * (C3) All `inputs` have the same shape.
+
+  * (C4) All `updates` have the same shape.
+
+  * (C5) $0 \le$ `index_vector_dim` $\le$ rank(`scatter_indices`).
+
+  * (C6) size(`scatter_dims_to_operand_dims`) $=$
+         `index_vector_dim` $\lt$ rank(`scatter_indices`) ?
+         dim(`scatter_indices`, `index_vector_dim`) : 1.
+
+  * (C7) All dimensions in `scatter_dims_to_operand_dims` unique.
+
+  * (C8) For all i $\in$ [0, size(`scatter_dims_to_operand_dims`)), $0 \le$
+        `scatter_dims_to_operand_dims`[i] $\lt$ rank(`operand`[k]), for any k
+        $\in$ [0, N).
+
+  * (C9) All dimensions in `update_window_dims` are unique and sorted.
+
+  * (C10) For all i $\in$ [0, size(`update_window_dims`)), $0 \le$
+    `update_window_dims`[i] $\lt$ rank(`updates`[k]), for any k $\in$ [0, N).
+
+  * (C11) All dimensions in `inserted_window_dims` are unique and sorted.
+
+  * (C12) For all i $\in$ [0, size(`inserted_window_dims`)), $0 \le$
+    `inserted_window_dims`[i] $\lt$ rank(`inputs`[k]), for any k $\in$ [0, N).
+
+  * (C13) `update_computation` has type `(tensor<E0>, ..., tensor<EN-1>, tensor<E0>, ..., tensor<EN-1>) -> (tensor<E0>, ..., tensor<EN-1>)`
+          where `Ek = element_type(inputs[k])` for any k $\in$ [0, N).
+
+ * (C14) rank(`updates[k]`) $=$ `effective_scatter_indices_rank` - 1 $+$
+         size(`update_window_dims`), where
+         `effective_scatter_indices_rank` $=$
+         `index_vector_dim` $\lt$ rank(`scatter_indices`) ?
+         rank(`scatter_indices`) : rank(`scatter_indices`) + 1.
+
+ * (C15) `shape(updates[k])` $=$ `concatenate(shape(scatter_indices), shape(inputs[k]))`
+         except that:
+   * The dimension size of `scatter_indices` corresponding to
+     `index_vector_dim` is not included.
+   * The dimension sizes in `inputs[k]` corresponding to
+     `inserted_window_dims` are not included.
```

Comment:
As it's currently written, C15 has two problems: 1) `concatenate` doesn't work because `update_scatter_dims` and `update_window_dims` don't have to be consecutive, 2) the part corresponding to `update_window_dims` is actually $\le$ and not $=$. Here's the rewording that fixes it:

```
 * (C15) `shape(updates[k])` $=$ 
         `combine(update_scatter_dim_sizes, update_window_dim_sizes)` where:
   * `update_scatter_dim_sizes` = `shape(scatter_indices)` except that
     the dimension size of `scatter_indices` corresponding to `index_vector_dim` 
     is not included.
   * `update_window_dim_sizes` $\le$ `shape(inputs[0])` except that 
     the dimension sizes in `inputs[0]` corresponding to `inserted_window_dims` 
     are not included.
   * `combine` puts `update_scatter_dim_sizes` at axes corresponding to 
     `update_scatter_dims` and `update_window_dim_sizes` at axes corresponding
     to `update_window_dims`.
```

Gather has the same problem, by the way. Here's the rewording that fixes it as well:

```
* (C14) `shape(result)` $=$ `combine(batch_dim_sizes, offset_dim_sizes)` where:
   * `batch_dim_sizes` = `shape(start_indices)` except that
     the dimension size of `start_indices` corresponding to `index_vector_dim` 
     is not included.
   * `offset_dim_sizes` $\le$ `shape(slice_sizes)` except that 
     the dimension sizes in `slice_sizes` corresponding to 
     `collapsed_slice_dims` are not included.
   * `combine` puts `batch_dim_sizes` at axes corresponding to `batch_dims` and 
     `offset_dim_sizes` at axes corresponding to `offset_dims`.
```